shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -764,27 +744,84 @@ public void sendPlatformMessage(String channel, String message, MessageReplyCall
      * @param message Message payload.
      * @param callback Callback that receives a reply from the application.
      */
-    public void sendToFlutter(String channel, String message, MessageReplyCallback callback) {
-        sendPlatformMessage(channel, message, callback);
+    public void sendToFlutter(String channel, String message, final MessageReplyCallback callback) {
+        sendToFlutter(channel, ByteBufferTools.extractBytes(message),
+            callback == null ? null : new BinaryMessageReplyCallback() {
+                @Override
+                public void onReply(ByteBuffer reply) {
+                    callback.onReply(ByteBufferTools.extractString(reply));
+                }
+            });
     }
 
     public void sendToFlutter(String channel, String message) {
         sendToFlutter(channel, message, null);
     }
 
-    /** Callback invoked when the app replies to a message sent with sendToFlutter. */
+    /**
+     * Send a binary message to the Flutter application.  The Flutter Dart code can register a
+     * platform message handler that will receive these messages.
+     * @param channel Name of the channel that will receive this message.
+     * @param message Message payload, a {@link ByteBuffer} with the message bytes between position
+     * zero and current position, or null.
+     * @param callback Callback that receives a reply from the application.
+     */
+    public void sendToFlutter(String channel, ByteBuffer message, BinaryMessageReplyCallback callback) {
```

Comment:
I'd stack these the other way so that they bottom out in `sendBinaryMessage` with the intent of removing `sendToFlutter` and `sendPlatformMessage` once we convert all the existing clients to using codecs.